### PR TITLE
Fixed hand posing when using custom mesh

### PR DESCRIPTION
**Pull request by intvoker**

### CODE REVIEW

File: Plugins/OculusHandTools/Source/OculusHandPoseRecognition/Private/PoseableHandComponent.cpp [hunk 1]
```diff
@@ -3,6 +3,17 @@
 #include "PoseableHandComponent.h"
 #include "OculusHandPoseRecognitionModule.h"
 
+void UPoseableHandComponent::BeginPlay()
+{
+	// Unfortunately, bCustomHandMesh in UOculusHandComponent is private
+	if (SkeletalMesh)
+	{
+		bCustomHandMesh = true;
```
Comment (jasonmeisel) [inline Plugins/OculusHandTools/Source/OculusHandPoseRecognition/Private/PoseableHandComponent.cpp]:
```suggestion
		bCustomPoseableHandMesh = true;
```

File: Plugins/OculusHandTools/Source/OculusHandPoseRecognition/Private/PoseableHandComponent.cpp [hunk 2]
```diff
@@ -40,7 +51,17 @@ void UPoseableHandComponent::TickComponent(float DeltaTime, enum ELevelTick Tick
 
 void UPoseableHandComponent::UpdateBonePose(EBone Bone, ERecognizedBone PosedBone)
 {
-	auto const BoneIndex = (int)Bone;
+	auto BoneIndex = (int)Bone;
+
+	if (bCustomHandMesh && BoneNameMappings.Contains(Bone))
```
Comment (jasonmeisel) [inline Plugins/OculusHandTools/Source/OculusHandPoseRecognition/Private/PoseableHandComponent.cpp]:
```suggestion
	if (bCustomPoseableHandMesh && BoneNameMappings.Contains(Bone))
```

File: Plugins/OculusHandTools/Source/OculusHandPoseRecognition/Private/PoseableHandComponent.cpp [hunk 3]
```diff
@@ -40,7 +51,17 @@ void UPoseableHandComponent::TickComponent(float DeltaTime, enum ELevelTick Tick
...
+	{
+		const auto MappedBoneIndex = SkeletalMesh->GetRefSkeleton().FindBoneIndex(BoneNameMappings[Bone]);
```
Comment (jasonmeisel) [inline Plugins/OculusHandTools/Source/OculusHandPoseRecognition/Private/PoseableHandComponent.cpp]:
```suggestion
		auto const MappedBoneIndex = SkeletalMesh->GetRefSkeleton().FindBoneIndex(BoneNameMappings[Bone]);
```

File: Plugins/OculusHandTools/Source/OculusHandPoseRecognition/Public/PoseableHandComponent.h [hunk 4]
```diff
@@ -25,6 +27,9 @@ class OCULUSHANDPOSERECOGNITION_API UPoseableHandComponent : public UOculusHandC
 	UPROPERTY(EditAnywhere, BlueprintReadWrite)
 	float DefaultLerpSpeed = 10.f;
 
+protected:
+	bool bCustomHandMesh = false;
```
Comment (jasonmeisel) [inline Plugins/OculusHandTools/Source/OculusHandPoseRecognition/Public/PoseableHandComponent.h]:
```suggestion
	bool bCustomPoseableHandMesh = false;
```

**Review by jasonmeisel — CHANGES_REQUESTED** [review]
This seems like a great fix!

We suggest you change `bCustomHandMesh` to `bCustomPoseableHandMesh`, so that it isn't confused with the private field of the base class.

Thanks for the contribution!

**Comment by intvoker** [comment]
@jasonmeisel, fixed!

**Review by AlexThiv — APPROVED** [review]

**Pull request merged**